animation_nodes/nodes/gpencil/gp_frame_from_strokes.py

```diff
@@ -0,0 +1,49 @@
+import bpy
+import numpy as np
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPFrame
+from ... base_types import AnimationNode, VectorizedSocket
+
+class GPFrameFromStrokesNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPFrameFromStrokesNode"
+    bl_label = "GP Frame From Strokes"
+    errorHandlingType = "EXCEPTION"
+
+    useStrokeList: VectorizedSocket.newProperty()
+    useNumberList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("GPStroke", "useStrokeList",
+            ("Stroke", "stroke"), ("Strokes", "strokes")), dataIsModified = True)
+        self.newInput(VectorizedSocket("Float", "useNumberList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "frameNumbers")), dataIsModified = True)
+        self.newOutput(VectorizedSocket("GPFrame", "useNumberList",
+            ("Frame", "frame"), ("Frames", "frames")))
+
+    def getExecutionFunctionName(self):
+        if self.useNumberList:
+            return "execute_StrokesFrameNumbers"
+        else:
+            return "execute_StrokesFrameNumber"
+
+    def execute_StrokesFrameNumber(self, strokes, frameNumber):
+        if not self.useStrokeList:
+            strokes = [strokes]
+
+        return GPFrame(strokes, 0, frameNumber)
+
+    def execute_StrokesFrameNumbers(self, strokes, frameNumbers):
+        frames = []
+        if not self.useStrokeList:
+            return frames
+
+        if len(strokes) > len(frameNumbers):
```

Comment:
What if it was less than?

Reply:
This is the one to one case i.e., each stroke has its own frame. However, if it is less than then we can assign the rest of the stroke on to the last frame.

Reply:
If I understand the function correctly. If the number of strokes is less than the number of frames. The rest of the frames will be ignored. Which doesn't seem predictable enough. So you should either use virtual lists here or raise an error.

Reply:
Okay, here I'll raise an error.